src/main/kotlin/net/atos/zac/identity/IdentityService.kt

```diff
@@ -60,4 +62,20 @@ class IdentityService @Inject constructor(
             .map { it.toUser() }
             .sortedBy { it.getFullName() }
     }
+
+    fun listGroupNamesForUser(userId: String): List<String> {
+        val keycloakUserId = keycloakZacRealmResource.users()
```

Comment:
We already have the `readUser()` function to retrieve the user. Why not use that? This is unnecessary duplicate code.

Reply:
ps: that `readUser()` function has issues we should also fix but one step at a time. ;-)

Reply:
`readUser` returns username and not keycloak user id we need to fetch the groups.

In theory search by username and full representation can solve this. In practice the admin library does not map the group as with https://github.com/keycloak/keycloak/discussions/23674

Reply:
ok clear!

Instead of a `IdentityRuntimeException` probably better to make an explicit `UserNotFoundException` (which extends from IdentityRuntimeException) with an error code which could be shown in the frontend but we can leave that for another day. :-)

Reply:
So in short - we cannot reuse `readUser()`. We need keycloak user ID and that's not available in `User`

Reply:
Added `UserNotFoundException` and `GroupNotFoundException`